src/libpriv/libdnf/dnf-version.h

```diff
@@ -0,0 +1,15 @@
+// This is an awful hack necessary because this header file
+// is generated in the libdnf build, but we don't want to serialize
+// our C++ build waiting for that.  So just generate it ahead
+// of time.
```

Comment:
So IIUC, this is intended to fulfill requests for `#include <libdnf/dnf-version.h>` when compiling rpm-ostree? So then theoretically if we use those macros somewhere in rpm-ostree, we'd pull in the wrong ones? I feel like this will come back to bite us in the future (or cause some subtle bug that'll be really hard to debug).

Hmm, how much is this hack buying us? Aren't we parallelizing the Rust builds anyway? For the developer case at least, I think it's OK if a fresh build takes a bit longer. And as we move more things from C++ to Rust, we'll naturally speed up C++ compilation. I'd vote for dropping this hack, but fine with it if others don't mind it.

Reply:
I think we could wire things up so that `make → cargo build -p libdnf-sys →C++` and in parallel `make -> cargo build`, because cargo will do locking internally.

But...another ugly part is that cargo puts the output in a target directory only known to the `build.rs` script, so we'd have to write a file with that and load it into `make` somehow.

If you look at e.g. `rg 'LIBDNF_.*VERSION'` the version macros are not used by us, and are basically not used by libdnf either.  The only actual usage in code is that they're exposed in the Python "hawkey" bindings which we don't even build.

Reply:
> If you look at e.g. `rg 'LIBDNF_.*VERSION'` the version macros are not used by us, and are basically not used by libdnf either. The only actual usage in code is that they're exposed in the Python "hawkey" bindings which we don't even build.

Yeah, saw that. I agree it's safe to do this now, but mocking build output just feels very underhanded. It's the kind of thing no one will notice until 3 years from now when someone comes upon it after debugging for 4 hours because something started relying on those macros not lying and breaking something else. (Offhand, I could also imagine `rpm-ostree --version` printing the version of libdnf that we're vendoring using these macros so that you don't have to do the lookup in GitHub.)

Anyway, fine as is too and maybe I'm over-estimating the likelihood of this causing trouble in the future. But again I'd vote for optimizing the build system in other ways instead.

Reply:
It looks like the values here currently come from the spec file https://github.com/rpm-software-management/libdnf/commit/a240bf275431fe19883f9f38572fca4d95b69545 so...it'd be quite ugly to replicate that, and in any case it's really going to be a lie anyways.

OK I changed things so that the header file is empty - that way if e.g. libdnf starts depending on the value in any way it becomes a hard compile error.

Another thing we could do I guess is validate the sha256 checksum of the vendored header copy and fail if it changed, but eh.